bpfd-operator/apis/v1alpha1/shared_types.go

```diff
@@ -102,3 +102,178 @@ type ImagePullSecretSelector struct {
 	// Namespace of the secret which contains the credentials to access the image repository.
 	Namespace string `json:"namespace"`
 }
+
+// -----------------------------------------------------------------------------
+// Status Conditions - BPF Programs
+// -----------------------------------------------------------------------------
+
+const (
+	// BpfdOperatorFinalizer is the finalizer that holds a BPF program from
+	// deletion until cleanup can be performed.
+	BpfdOperatorFinalizer = "bpfd.io.operator/finalizer"
```

Comment:
Should we put this in https://github.com/bpfd-dev/bpfd/blob/main/bpfd-operator/internal/constants.go#L21? If not we should also add all these controller specific finalizers to the API.